### PR TITLE
refactor!: add an `Scm` abstraction

### PR DESCRIPTION
**Pull request by chingor13**
Rather than assuming we are using `GitHub` create an `Scm` interface that abstracts the git source. This is useful for injecting a fake SCM for testing or for potentially supporting other SCM systems in the future. We do not have immediate plans to implement any other SCMs, but this refactor allows us to future-proof this.

This is a breaking change because we have renamed a few (unlikely to be used, but still technically public) options parameters from `github` to `scm`. We also have renamed a few types to change `GitHub` to `Scm` (e.g. `GitHubRelease` -> `ScmRelease`).

This is mostly just a trivial (renaming) pull request that I'd like to include in the next major version (when we drop node 12 support).

We also change the `skip-github-release` manifest option to `skip-release` although both options are still supported.

### CODE REVIEW

**Comment by chingor13** [comment]
Note: this is merging into a v14-next branch which will be for the v14 major version

**Review by bcoe — APPROVED** [review]
This seems like a good incremental refactor.

I assume the value to this, is that we'd have an Scm instance that uses the git cli for some operations, the GitHub API for others, reducing the overall load on quota?

**Review by bcoe — COMMENTED** [review]
I think it would be worth getting one more reviewer for this, since it sets the stage for significant future work.

File: __snapshots__/php-composer-update-packages.js
Comment (chingor13) [reply __snapshots__/php-composer-update-packages.js:9]: Yeah, I think these were excess from an external contributor's PR

File: src/bin/release-please.ts
Comment (chingor13) [reply src/bin/release-please.ts:19]: I'm ok with the edges of the code knowing about the concrete implementation of the SCM. The binary script here still assumes GitHub is the SCM as it's the only one. In the future, this would use a factory to build the SCM from config.

File: src/scm.ts
Comment (chingor13) [reply src/scm.ts:140]: Whatever SCM we build would need to have a notion of pull requests (e.g. a change set in gerritt). Otherwise, release-please fundamentally wouldn't work -- it needs to create a release pull request of some kind.

**Comment by SurferJeffAtGoogle** [comment]
Whenever I've tried to create an abstraction for a concrete service, when the second implementation of the service actually arrived it violated all my assumptions and I had to start over.  So, in short, I don't think we should attempt to make the SCM abstract until we have a second SCM ready to be plugged in.

**Comment by chingor13** [comment]
> it violated all my assumptions and I had to start over

I agree

> attempt to make the SCM abstract until we have a second SCM ready to be plugged in

I think this refactor might be too daunting for an external contributor to attempt if they wanted to contribute a new SCM (like Gitlab or gerrit). My reason for proactively creating it is to unblock someone from attempting it.

**Pull request closed without merging**
